### PR TITLE
Pay toggle per player

### LINKED ISSUE #3087: /paytoggle <Player> on/off
### Feature request

**Feature description**
/paytoggle <Player> on/off

**How the feature is useful**
Allow admins take control of who can receive payments if they dont have permissions to use the /paytoggle command

### PR DESCRIPTION
**Pull request by TreyRuffy**
Allows the player to do /paytoggle <player> to only toggle pay for that single player

Closes #3087

### CODE REVIEW

**Comment by TreyRuffy** [comment]
This is for #3087. Tell me if there's anything I overlooked, and I should be able to fix it.

File: Essentials/src/com/earth2me/essentials/User.java [hunk 1]
```diff
@@ -929,6 +929,11 @@ public long getAfkSince() {
         return afkSince;
     }
 
+    @Override
+    public boolean isPayBlockExempt() {
+        return this.isAuthorized("essentials.paytoggle.exempt");
```
Comment (pop4959) [inline Essentials/src/com/earth2me/essentials/User.java:934]: Is this necessary? You're only checking the permission once, in the command.

File: Essentials/src/messages_en.properties [hunk 2]
```diff
@@ -353,6 +353,7 @@ noMetaJson=JSON Metadata is not supported in this version of Bukkit.
 noMetaPerm=§4You do not have permission to apply §c{0}§4 meta to this item.
 none=none
 noNewMail=§6You have no new mail.
+noPayBlocked=§6You are not blocking anyone''s pay.
```
Comment (pop4959) [inline Essentials/src/messages_en.properties:356]: Messages should be in messages.properties only, as the localized versions will be overwritten by Crowdin.

File: Essentials/src/com/earth2me/essentials/commands/Commandpaytoggle.java [hunk 3]
```diff
@@ -15,17 +15,48 @@ public Commandpaytoggle() {
...
+            }
+            user.setAcceptingPay(acceptingPay);
+            if (acceptingPay) {
+                user.sendMessage(tl("payToggleOn"));
+            } else {
+                user.sendMessage(tl("payToggleOff"));
+            }
         } else {
-            user.sendMessage(tl("payToggleOff"));
+            if (args[0].equalsIgnoreCase("players")) {
```
Comment (pop4959) [inline Essentials/src/com/earth2me/essentials/commands/Commandpaytoggle.java:32]: I don't see why this list feature is necessary. Why not just tell the player when the payment is going through that it's blocked?

File: Essentials/src/com/earth2me/essentials/commands/Commandpaytoggle.java [hunk 4]
```diff
@@ -15,17 +15,48 @@ public Commandpaytoggle() {
...
+                for (String s : user._getBlockingIndividualPay()) {
+                    sb.append(s).append(" ");
+                }
+                String payBlockedList = sb.toString().trim();
+                user.sendMessage(payBlockedList.length() > 0 ? tl("payBlockedList", payBlockedList) : tl(
+                        "noPayBlocked"));
+            } else {
+                User player;
+                try {
+                    player = getPlayer(server, args, 0, true, true);
```
Comment (pop4959) [inline Essentials/src/com/earth2me/essentials/commands/Commandpaytoggle.java:43]: getPlayer already gets offline players, this try/catch thing isn't necessary.

File: Essentials/src/com/earth2me/essentials/commands/Commandpaytoggle.java [hunk 5]
```diff
@@ -15,17 +15,48 @@ public Commandpaytoggle() {
...
+                } catch (PlayerNotFoundException ex) {
+                    player = ess.getOfflineUser(args[0]);
+                }
+                if (player == null) {
+                    throw new PlayerNotFoundException();
```
Comment (pop4959) [inline Essentials/src/com/earth2me/essentials/commands/Commandpaytoggle.java:48]: Don't catch and re-throw this. If getPlayer doesn't find it, just let it throw itself.

**Review by pop4959 — CHANGES_REQUESTED** [review]
Not a full review, but a few comments to get you started. Also feels like you could clean this up a bit more.

**Comment by mdcfe** [comment]
This PR has nothing to do with #3087. As far as I can tell, #3087 is asking for `/paytoggle` to function as a `EssentialsToggleCommand` like `/msgtoggle` and `/vanish`, which is reasonable (not sure why it wasn't implemented this way in the first place).

In contrast, this PR adds a secondary function to `/paytoggle` that allows players to block specific players from `/pay`. I'm not sure what the use case for this feature is? `/pay` could just be designed to respect `/ignore`, which would be a lot less convoluted as that way players don't need to manage several block lists.

**Comment by pop4959** [comment]
Closing this in favor of #3273, as it seems this PR is inactive

**Pull request closed without merging**